bridges/snowbridge/pallets/system/src/lib.rs

```diff
@@ -574,6 +592,29 @@ pub mod pallet {
 			});
 			Ok(())
 		}
+
+		/// Sends a message to the Gateway contract to register a new
+		/// token that represents `asset`.
+		///
+		/// - `origin`: Must be `MultiLocation` from sibling parachain
+		#[pallet::call_index(10)]
+		#[pallet::weight(T::WeightInfo::register_token())]
+		pub fn register_token(
+			origin: OriginFor<T>,
+			asset: Box<VersionedLocation>,
+			metadata: AssetRegistrarMetadata,
```

Comment:
I think we can pass the inner metadata fields directly. Is there any benefit to having them in a struct?

Reply:
Somehow follows 
https://github.com/moonbeam-foundation/moonbeam/blob/80d6f17ad88252b772b0bd35064e2392494473e0/pallets/asset-manager/src/lib.rs#L76 as cohesive unit.

Reply:
Let's at least rename the struct to AssetMetadata

Reply:
https://github.com/Snowfork/polkadot-sdk/pull/128/commits/79e62fa1e8b326fcec6c8ae40973ed1de94c92b9